src/util.c

```diff
@@ -49,6 +49,57 @@ char *path_append(const char *path, const char *filename)
     return str;
 }
 
+char *escape_char(const char *url, const ESCAPE_CHAR c)
+{
+    static char escaped[MAX_PATH_LEN];
```

Comment:
Have you seen this pattern being used elsewhere in this code base? If so, please flag then up. 

And no, please don't do this. I don't think this is threadsafe. Another thread might come in and call this function again, while this function is already called.

Reply:
https://github.com/fangfufu/httpdirfs/pull/137/commits/bedfdf28afa75040a624ddb7834b28fcdf952fab